overlord/devicestate/devicestate_gadget_test.go

```diff
@@ -208,7 +209,34 @@ func (s *deviceMgrGadgetSuite) setupUC20ModelWithGadget(c *C, gadget string) {
 	})
 }
 
-func (s *deviceMgrGadgetSuite) setupGadgetUpdate(c *C, modelGrade, gadgetYamlContent, gadgetYamlContentNext string) (chg *state.Change, tsk *state.Task) {
+func (s *deviceMgrGadgetSuite) setupClassicWithModesModel(c *C, gadget string) *asserts.Model {
```

Comment:
There is a very similar helper defined above in overlord/devicestate/devicestate_bootconfig_test.go - I wonder if we should consolidate into something like - but probably something for later as there are already some setup* helpers that would also need to be considered.
```go
diff --git a/overlord/devicestate/devicestate_bootconfig_test.go b/overlord/devicestate/devicestate_bootconfig_test.go
index f3930ab104..c53b277d3b 100644
--- a/overlord/devicestate/devicestate_bootconfig_test.go
+++ b/overlord/devicestate/devicestate_bootconfig_test.go
@@ -99,33 +99,6 @@ func (s *deviceMgrBootconfigSuite) setupUC20Model(c *C) *asserts.Model {
        return s.makeModelAssertionInState(c, "canonical", "pc-model-20", mockCore20ModelHeaders)
 }
 
-func (s *deviceMgrBootconfigSuite) setupClassicWithModesModel(c *C) *asserts.Model {
-       devicestatetest.SetDevice(s.state, &auth.DeviceState{
-               Brand:  "canonical",
-               Model:  "classic-with-modes",
-               Serial: "didididi",
-       })
-       return s.makeModelAssertionInState(c, "canonical", "classic-with-modes",
-               map[string]interface{}{
-                       "architecture": "amd64",
-                       "classic":      "true",
-                       "distribution": "ubuntu",
-                       "base":         "core22",
-                       "snaps": []interface{}{
-                               map[string]interface{}{
-                                       "name": "pc-linux",
-                                       "id":   "pclinuxdidididididididididididid",
-                                       "type": "kernel",
-                               },
-                               map[string]interface{}{
-                                       "name": "pc",
-                                       "id":   "pcididididididididididididididid",
-                                       "type": "gadget",
-                               },
-                       },
-               })
-}
-
 func (s *deviceMgrBootconfigSuite) testBootConfigUpdateRun(c *C, updateAttempted, applied bool, errMatch string) {
        restore := release.MockOnClassic(false)
        defer restore()
@@ -230,7 +203,7 @@ func (s *deviceMgrBootconfigSuite) TestBootConfigUpdateRunSuccess(c *C) {
 
 func (s *deviceMgrBootconfigSuite) TestBootConfigUpdateRunSuccessClassic(c *C) {
        s.state.Lock()
-       s.setupClassicWithModesModel(c)
+       s.setupClassicWithModesModel(c, "pc")
        s.state.Unlock()
 
        s.managedbl.Updated = true
diff --git a/overlord/devicestate/devicestate_gadget_test.go b/overlord/devicestate/devicestate_gadget_test.go
index ee68b5c923..34831abcb8 100644
--- a/overlord/devicestate/devicestate_gadget_test.go
+++ b/overlord/devicestate/devicestate_gadget_test.go
@@ -209,33 +209,6 @@ func (s *deviceMgrGadgetSuite) setupUC20ModelWithGadget(c *C, gadget string) {
        })
 }
 
-func (s *deviceMgrGadgetSuite) setupClassicWithModesModel(c *C, gadget string) *asserts.Model {
-       devicestatetest.SetDevice(s.state, &auth.DeviceState{
-               Brand:  "canonical",
-               Model:  "classic-with-modes",
-               Serial: "didididi",
-       })
-       return s.makeModelAssertionInState(c, "canonical", "classic-with-modes",
-               map[string]interface{}{
-                       "architecture": "amd64",
-                       "classic":      "true",
-                       "distribution": "ubuntu",
-                       "base":         "core22",
-                       "snaps": []interface{}{
-                               map[string]interface{}{
-                                       "name": "pc-linux",
-                                       "id":   "pclinuxdidididididididididididid",
-                                       "type": "kernel",
-                               },
-                               map[string]interface{}{
-                                       "name": gadget,
-                                       "id":   "pcididididididididididididididid",
-                                       "type": "gadget",
-                               },
-                       },
-               })
-}
-
 func (s *deviceMgrGadgetSuite) setupGadgetUpdate(c *C, modelGrade, gadgetYamlContent, gadgetYamlContentNext string, isClassic bool) (chg *state.Change, tsk *state.Task) {
        siCurrent := &snap.SideInfo{
                RealName: "foo-gadget",
diff --git a/overlord/devicestate/devicestate_test.go b/overlord/devicestate/devicestate_test.go
index 824d2e6d50..6edc2756c5 100644
--- a/overlord/devicestate/devicestate_test.go
+++ b/overlord/devicestate/devicestate_test.go
@@ -356,6 +356,33 @@ func (s *deviceMgrBaseSuite) makeSerialAssertionInState(c *C, brandID, model, se
        return makeSerialAssertionInState(c, s.brands, s.state, brandID, model, serialN)
 }
 
+func (s *deviceMgrBaseSuite) setupClassicWithModesModel(c *C, gadget string) *asserts.Model {
+       devicestatetest.SetDevice(s.state, &auth.DeviceState{
+               Brand:  "canonical",
+               Model:  "classic-with-modes",
+               Serial: "didididi",
+       })
+       return s.makeModelAssertionInState(c, "canonical", "classic-with-modes",
+               map[string]interface{}{
+                       "architecture": "amd64",
+                       "classic":      "true",
+                       "distribution": "ubuntu",
+                       "base":         "core22",
+                       "snaps": []interface{}{
+                               map[string]interface{}{
+                                       "name": "pc-linux",
+                                       "id":   "pclinuxdidididididididididididid",
+                                       "type": "kernel",
+                               },
+                               map[string]interface{}{
+                                       "name": gadget,
+                                       "id":   "pcididididididididididididididid",
+                                       "type": "gadget",
+                               },
+                       },
+               })
+}
+
 func (s *deviceMgrSuite) SetUpTest(c *C) {
        classic := false
        s.setupBaseTest(c, classic)
```

```